### ISSUE TITLE
Overlaying imshow with transparancy / alpha yields different results in pdf and png when using savefig

### ISSUE THREAD

**Issue by TAdeJong** [matplotlib/matplotlib#18871]
### Bug report

**Bug summary**

When using multiple calls to `plt.imshow()` with transparancy/alpha on the top image, the .pdf and .png versions are inconsistent

**Code for reproduction**

Code for reproduction below, also available as a binder at https://mybinder.org/v2/gist/TAdeJong/8cb84dc74bc6d2c2a1744ef465bf85d8/HEAD . 
Effect can be even more noticable when overlaying actual images, but for simplicity this example keeps it a single colors:

```python
import matplotlib.pyplot as plt
import matplotlib
import numpy as np

print(matplotlib.__version__, matplotlib.get_backend())

fig,ax = plt.subplots(ncols=11, figsize=[12,3])
for i in range(11):
    ax[i].imshow(np.full((100,100), 0.1), cmap='gray', vmin=0, vmax=1)
    ax[i].imshow(np.full((100,100), 0.9), cmap='inferno', vmin=0, vmax=1, alpha=0.1*i)
    ax[i].set_title(f"$\\alpha=${0.1*i:.1f}")
plt.tight_layout()
plt.savefig('test.png')
plt.savefig('test.pdf')
```

**Actual outcome**

Output files:
![image](https://user-images.githubusercontent.com/6816964/97873610-2dd0a080-1d18-11eb-81da-a100231a811f.png)
Colors in output files seem consistent between different OS and viewers.

```
# Mybinder version info:
3.3.2 module://ipykernel.pylab.backend_inline
# Local version info:
3.3.2 TkAgg

```

**Expected outcome**

Produced colors being consistent between different output formats, i.e. a equal color for each of the respective squares between the PNG and PDF  output.
**OR** a clear warning in the documentation (I am not aware of such a warning existing, but I might have missed it).

**Matplotlib version**

  * Operating system: Ubuntu/mybinder
  * Matplotlib version: 3.3.2, at least down to 3.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline / TkAgg
  * Python version: 3.7.7, 3.7.3, 3.7.8
  * Jupyter version (if applicable):
  * Other libraries: 

#Edit: fix mybinder link.

**Comment by anntzer** [comment]
Possibly related to https://github.com/matplotlib/matplotlib/issues/8847?

**Comment by TAdeJong** [comment]
Yes, they do seem related! Using the code from #8847 and save as pdf and png yields the same images:
![image](https://user-images.githubusercontent.com/6816964/97880795-e3542180-1d21-11eb-8c56-0c31fde399ff.png)

**Comment by TAdeJong** [comment]
For those stumbling upon this problem: I found a workaround, as the .pdf get's the same behaviour as the .png when adding `interpolation='none'` to at least one of the initial imshow calls, like such:

```python
fig,ax = plt.subplots(ncols=11, figsize=[12,3])
for i in range(11):
    ax[i].imshow(np.full((100,100), 0.1), cmap='gray', vmin=0, vmax=1, interpolation='none')
    ax[i].imshow(np.full((100,100), 0.9), cmap='inferno', vmin=0, vmax=1, alpha=0.1*i)#, interpolation='none')
    ax[i].set_title(f"$\\alpha=${0.1*i:.1f}")
plt.tight_layout()
plt.savefig('test_interp.png')
plt.savefig('test_interp.pdf')
```

**Comment by codeceptsDE** [comment]
wow, thanks @TAdeJong for the workaround! 
The issue still persists, but it seems like the referenced https://github.com/matplotlib/matplotlib/issues/8847 just recently received a fix PR https://github.com/matplotlib/matplotlib/pull/32107. 
Lets see what effect this will have on this issue, and also on #25806!

**Comment by ayshih** [comment]
I've confirmed that this issue has also been fixed by #32107
